### ISSUE TITLE
Not all new messages are shown on the subscriptions tab

### ISSUE THREAD

**Issue by yurivict** [Bitmessage/PyBitmessage#856]
'Subscriptions' tab shows that there are new messages (in bold), but not all new messages are displayed there. 'Messages' tab though shows all new messages from the subscription channels.

**Comment by PeterSurda** [comment]
I'm not sure I understand. Can you send me a screenshot maybe? That being said, there are two bugs that could be related:

- when you start, no subscription/chan is selected and all of them are shown
- when you read a message in "all" or under subscription/chan, the list is not updated in the other one

But that does not sound like what you're describing.

**Comment by yurivict** [comment]
1. Subscription's "inbox" shows 1 new message (in bold).
2. When I click on the inbox this new message isn't displayed in the list of messages to the right.
3. To see the message I have to switch to the "Messages" tab where it is shown under "All account"

**Comment by PeterSurda** [comment]
Could it be that you have something in the search field?

**Comment by yurivict** [comment]
No, search field is empty.

**Comment by PeterSurda** [comment]
It's likely that this was a bug in the indicator of the number of new messages (there have been many over the time and maybe there still is one) rather than a bug in the list of messages in the inbox. If that's the case, the message would have been displayed, but it wouldn't have been bold. Is it possible?

**Comment by yurivict** [comment]
No, there are new messages that are received for the subscription, indicator shows it right. But they aren't in the list of this particular subscription. Some messages don't show under their subscriptions.

**Comment by PeterSurda** [comment]
Ok I will check.

**Comment by yurivict** [comment]
I have an additional info. This particular subscription is added in both Chans and Subscriptions tabs. (Not sure if this is supposed to be allowed)

"All accounts" list seems correct. "Subscriptions" tab correctly shows the count of new messages in bold (Inbox (N)) but the list is missing many of them. "Chans" shows the correct count and list but all of them are marked as "unread", regardless if they have been read on another tabs.

So, there are two problems:
1. "Subscriptions" tab is missing messages in the list
2. "Chans" tab has "unread" flag set wrongfully on the already read messages.

Additionally, it isn't clear to me if the same item should be allowed to be added to both Subscriptions and Chans. Not sure even what is the difference.

**Comment by PeterSurda** [comment]
I see. That would explain the misbehaviour. While technically it is possible to use the same key for both a subscription and a chan, it makes little practical sense and will very likely screw up the UI. The reason it makes little practical sense is that it results in a combination of disadvantages of both mechanisms: since all the chan subscribers know the private key, they all can send broadcasts with that key, which makes the broadcast author unauthenticated, which makes no sense. I could change the UI so that it's displayed correctly but it's too much work for a useless situation. I'm too lazy to check, but most likely the lists of messages are correct, just the indicator of unread message count is wrong.

If you are expecting to see chan messages under subscriptions, it means is a broader UI issue, i.e. it didn't explain to you the difference between chans and subscriptions. I also didn't understand it when I started working on it and I actually broke it in early releases of the new UI. There is a ticket open about explaining how it works to the user.

**Comment by yurivict** [comment]
If this is impractical you should make UI not allow to add the same item under both Subscriptions and Chans with an explanation why and possible choices.

**Comment by PeterSurda** [comment]
I believe it won't allow you to add a duplicate already, you probably imported the setup from an earlier version.

**Comment by mirrorwish** [comment]
What about renaming the subscriptions tab to "Broadcasts"? That would make it more clear.

**Comment by yurivict** [comment]
Possibly. But it is difficult to understand what the difference between them is. I think there should be notes on the tabs explaining the difference. Because chans are also broadcasts.

**Comment by PeterSurda** [comment]
I actually did some changes about this in https://github.com/Bitmessage/PyBitmessage/commit/6dff105a5b771c443d5ffbd5372a076fdb759712, can you check out the current v0.6 branch if it helps?
